sw/device/lib/dif/dif_aes.c

```diff
@@ -235,6 +235,23 @@ dif_result_t dif_aes_start(const dif_aes_t *aes,
     return result;
   }
 
+  if (transaction->mode == kDifAesModeGcm) {
+    // Set number of key bytes.
+    size_t key_bytes = 16;
+    if (transaction->key_len == kDifAesKey192) {
+      key_bytes = 24;
+    } else if (transaction->key_len == kDifAesKey256) {
+      key_bytes = 32;
+    }
+
+    uint32_t reg = bitfield_field32_write(
+        0, AES_CTRL_GCM_SHADOWED_NUM_VALID_BYTES_FIELD, key_bytes);
```

Comment:
What's the idea behind this? The [spec](https://docs.google.com/document/d/1CQMAL9Sgb7258iGOzDrz0KmZWY6pOvbYaAoX5iyqjFY/edit?tab=t.0) doesn't mention to configure the NUM_VALID_BYTES to the key length.

The description of the field says:
> CTRL_GCM_SHADOWED.NUM_VALID_BYTES: Indicates the number of valid bytes. Only the last block of the AAD and the plaintext/ciphertext are expected to have not all bytes valid. For all other blocks, the number of valid bytes should be set to 16.

So it should always be 16 for the initialization.